nav2_planner/src/planner_server.cpp

```diff
@@ -0,0 +1,251 @@
+PlannerServer::on_error(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_FATAL(get_logger(), "Lifecycle node entered error state");
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+PlannerServer::on_shutdown(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(get_logger(), "Shutting down");
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+void
+PlannerServer::computePathToPose()
+{
+  // Initialize the ComputePathToPose goal and result
+  auto goal = action_server_->get_current_goal();
+  auto result = std::make_shared<nav2_msgs::action::ComputePathToPose::Result>();
+
+  try {
+    if (action_server_ == nullptr) {
+      RCLCPP_DEBUG(get_logger(), "Action server unavailable. Stopping.");
+      return;
+    }
+
+    if (!action_server_->is_server_active()) {
+      RCLCPP_DEBUG(get_logger(), "Action server is inactive. Stopping.");
+      return;
+    }
+
+    if (action_server_->is_cancel_requested()) {
+      RCLCPP_INFO(get_logger(), "Goal was canceled. Canceling planning action.");
+      action_server_->terminate_goals();
+      return;
+    }
+
+    geometry_msgs::msg::PoseStamped start;
+    if (!nav2_util::getCurrentPose(start, *tf_)) {
+      return;
+    }
+
+    if (action_server_->is_preempt_requested()) {
+      RCLCPP_INFO(get_logger(), "Preempting the goal pose.");
+      goal = action_server_->accept_pending_goal();
+    }
+
+    RCLCPP_DEBUG(get_logger(), "Attempting to a find path from (%.2f, %.2f) to "
+      "(%.2f, %.2f).", start.pose.position.x, start.pose.position.y,
+      goal->pose.pose.position.x, goal->pose.pose.position.y);
+
+    result->path = planner_->createPlan(start, goal->pose);
+
+    if (result->path.poses.size() == 0) {
+      RCLCPP_WARN(get_logger(), "Planning algorithm %s failed to generate a valid"
+        " path to (%.2f, %.2f)", planner_plugin_name_.c_str(),
+        goal->pose.pose.position.x, goal->pose.pose.position.y);
+      // TODO(orduno): define behavior if a preemption is available
+      action_server_->terminate_goals();
+      return;
+    }
+
+    RCLCPP_DEBUG(get_logger(),
+      "Found valid path of size %u to (%.2f, %.2f)",
+      result->path.poses.size(), goal->pose.pose.position.x,
+      goal->pose.pose.position.y);
+
+    // Publish the plan for visualization purposes
+    RCLCPP_DEBUG(get_logger(), "Publishing the valid path");
+    publishPlan(result->path);
+
+    action_server_->succeeded_current(result);
+    return;
+  } catch (std::exception & ex) {
+    RCLCPP_WARN(get_logger(), "%s plugin failed to plan calculation to (%.2f, %.2f): \"%s\"",
+      planner_plugin_name_.c_str(), goal->pose.pose.position.x,
+      goal->pose.pose.position.y, ex.what());
+
+    // TODO(orduno): provide information about fail error to parent task,
+    //               for example: couldn't get costmap update
+    action_server_->terminate_goals();
+    return;
+  } catch (...) {
+    RCLCPP_WARN(get_logger(), "Plan calculation failed");
```

Comment:
Catching (...) should never happen since we've caught everything derived from std::exception up above. This should at least be an ERROR, and maybe a FATAL. I feel like stronger error wording is in order as well. Something like "An unexpected error has occurred. The planner server may not be able to continue operating correctly"

Reply:
I didn't add this, this was preexisting code, but I can make it more descriptive

Reply:
Also right above it, it does the catch with std::exception, this was here before I touched it, I think its good defensive programming in case someone decides to make a planner that throws exceptions non-derived from std::exception since its a plugin and people can do whatever they want

Reply:
I added the warning, but left the catch(...)